From 674c91b392c7f07c1c16fdd0d74cabb9b7a02117 Mon Sep 17 00:00:00 2001
From: Fei Jiang <fei.jiang@intel.com>
Date: Fri, 2 Feb 2018 21:21:12 +0800
Subject: [PATCH 0658/1240] drm/i915/gvt: resolve scheduler problem when run
 two UOS

When two UOS both run glmark, one UOS will stop every several seconds,
which is because the scheduler policy in GVT-g.
This fix includes:
1. Revert GVT_DEFAULT_TIME_SLICE to 1000000 to match
GVT_TS_BALANCE_PERIOD_MS and define GVT_TS_BALANCE_STAGE_NUM.
2. Set scheduler->next_vgpu[ring_id] to idle_vgpu when vgpu is idle.

Signed-off-by: Fei Jiang <fei.jiang@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>
---
 drivers/gpu/drm/i915/gvt/sched_policy.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/gvt/sched_policy.c b/drivers/gpu/drm/i915/gvt/sched_policy.c
index 4dbb068..f312da5 100644
--- a/drivers/gpu/drm/i915/gvt/sched_policy.c
+++ b/drivers/gpu/drm/i915/gvt/sched_policy.c
@@ -191,7 +191,7 @@ static struct intel_vgpu *find_busy_vgpu(struct gvt_sched_data *sched_data,
 }
 
 /* in nanosecond */
-#define GVT_DEFAULT_TIME_SLICE 5000000
+#define GVT_DEFAULT_TIME_SLICE 1000000
 
 static void tbs_sched_func(struct gvt_sched_data *sched_data, enum intel_engine_id ring_id)
 {
@@ -214,6 +214,8 @@ static void tbs_sched_func(struct gvt_sched_data *sched_data, enum intel_engine_
 				&sched_data->lru_runq_head[ring_id]);
 
 		gvt_dbg_sched("pick next vgpu %d\n", vgpu->id);
+	} else {
+		scheduler->next_vgpu[ring_id] = gvt->idle_vgpu;
 	}
 out:
 	if (scheduler->next_vgpu[ring_id]) {
-- 
2.7.4

